### ISSUE TITLE
Pipeline Support

### ISSUE THREAD

**Issue by cbron** [rancher/rio#891]
Investigate Argo or Tekton for Rio pipelines, as well as user defined pipelines in riofile. 

related: https://github.com/rancher/rio/issues/377

**Comment by Vad1mo** [comment]
I have been looking into jenkins-x: they switched their previous jenkinsfile groov pipelines to [yaml ](https://jenkins-x.io/docs/concepts/jenkins-x-pipelines/)
> Jenkins X Pipelines, a new serverless pipeline execution engine based on the Tekton Pipelines 

I find jenkins-x complex, but I like the concept of jenkinsx.yml and its translation to Tekton Pipelines.

It would be good if riofile would translate to Tekton Pipelines but also provide the option to opt-out and hack a Tekton Pipeline if needed, for complexer usescases. Analogue to the current state with riofile and use of k8s deployment descriptors.

**Comment by cbron** [comment]
@Vad1mo yes that is our goal, to integrate with Riofile and still allow writing custom pipelines outside of Rio. We may also allow switching between pipeline vendors in the future.
